bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -1341,9 +1348,21 @@ public void operationComplete(Future<Channel> future) throws Exception {
                             }
                         } else if (future.isSuccess() && state == ConnectionState.START_TLS) {
                             rc = BKException.Code.OK;
-                            LOG.info("Successfully connected to bookie using TLS: " + addr);
+                            LOG.info("Successfully connected to bookie using TLS: {}", addr);
 
                             state = ConnectionState.CONNECTED;
+
+                            // print peer credentials
+                            SslHandler sslHandler = (SslHandler) channel.pipeline().get(SslHandler.class);
+                            LOG.info("Session {} is protected by {} ", future.get(),
+                                    sslHandler.engine().getSession().getCipherSuite());
+
+                            Certificate[] certificates = sslHandler.engine().getSession().getPeerCertificates();
+                            if (LOG.isDebugEnabled()) {
+                                LOG.debug("Peer Certificate Chain for {}: {} ", future.get(),
```

Comment:
Yes, in this specific case future is complete and successful. You are correct that if the future is not complete, it can throw an exception (I have seen this in the past).

Reply:
@eolivelli that's fine since this is executed in `future.isSuccess()` branch.